pallets/ocex/src/tests.rs

```diff
@@ -20,113 +20,605 @@ use frame_support::{
+		);
+	});
 }
+
+#[test]
+fn test_deposit_unknown_asset(){
+	let account_id = create_account_id();
+	new_test_ext().execute_with(|| {
+		assert_noop!(
+			OCEX::deposit(
+				Origin::signed(account_id.clone().into()),
+				AssetId::asset(10),
+				100_u128.into()
+			),
+			TokenError::UnknownAsset
+		);
+	});
+}
+
+#[test]
+fn test_deposit(){
+	let account_id = create_account_id();
+	new_test_ext().execute_with(||{
+		mint_into_account(account_id.clone());
+		assert_ok!(
+			OCEX::deposit(
+				Origin::signed(account_id.clone().into()),
+				AssetId::polkadex,
+				100_u128.into()
+			)
+		);
+	});
+}
+
+#[test]
+fn test_open_trading_pair_both_assets_cannot_be_same(){
+	new_test_ext().execute_with(||{
+		assert_noop!(
+			OCEX::open_trading_pair(
+				Origin::root(),
+				AssetId::asset(10),
+				AssetId::asset(10)
+			),
+			Error::<Test>::BothAssetsCannotBeSame
+		);
+	});
+}
+
+#[test]
+fn test_open_trading_pair_trading_pair_not_found(){
+	new_test_ext().execute_with(|| {
+		assert_noop!(
+			OCEX::open_trading_pair(
+				Origin::root(), 
+				AssetId::asset(10), 
+				AssetId::asset(20)
+			),
+			Error::<Test>::TradingPairNotFound
+		);
+	});
+}
+
+#[test]
+fn test_open_trading_pair(){
+	new_test_ext().execute_with(||{
+		assert_ok!(
+			OCEX::register_trading_pair(
+				Origin::root(), 
+				AssetId::asset(10), 
+				AssetId::asset(20), 
+				1_u128.into(),
+				100_u128.into(), 
+				1_u128.into(), 
+				100_u128.into(),
+				100_u128.into(),
+				10_u128.into()
+			)
+		);
+		assert_ok!(
+			OCEX::open_trading_pair(
+				Origin::root(),
+				AssetId::asset(10),
+				AssetId::asset(20)
+			)
+		);
+		assert_eq!(
+			TradingPairsStatus::<Test>::get(AssetId::asset(10), AssetId::asset(20)), 
+			true
+		);
+	})
+}
+
+#[test]
+fn test_close_trading_pair_both_assets_cannot_be_same(){
+	new_test_ext().execute_with(||{
+		assert_noop!(
+			OCEX::close_trading_pair(
+				Origin::root(),
+				AssetId::asset(10),
+				AssetId::asset(10)
+			),
+			Error::<Test>::BothAssetsCannotBeSame
+		);
+	});
+}
+
+#[test]
+fn test_close_trading_trading_pair_not_found(){
+	new_test_ext().execute_with(||{
+		assert_noop!(
+			OCEX::close_trading_pair(
+				Origin::root(),
+				AssetId::asset(10),
+				AssetId::asset(20)
+			),
+			Error::<Test>::TradingPairNotFound
+		);
+	});
+}
+
+#[test]
+fn test_close_trading_pair(){
+	new_test_ext().execute_with(||{
+		assert_ok!(
+			OCEX::register_trading_pair(
+				Origin::root(), 
+				AssetId::asset(10), 
+				AssetId::asset(20), 
+				1_u128.into(),
+				100_u128.into(), 
+				1_u128.into(), 
+				100_u128.into(),
+				100_u128.into(),
+				10_u128.into()
+			)
+		);
+		assert_ok!(
+			OCEX::close_trading_pair(
+				Origin::root(),
+				AssetId::asset(10),
+				AssetId::asset(20)
+			)
+		);
+		assert_eq!(
+			TradingPairsStatus::<Test>::get(AssetId::asset(10), AssetId::asset(20)), 
+			false
+		);
+	})
+}
+
+#[test]
+fn collect_fees(){
+	let account_id = create_account_id();
```

Comment:
I think should be added for each method test where `ensure_signed` returns error with `BadOrigin`.

Reply:
Updated ✅